apps/api/src/app/inbox/usecases/session/session.usecase.ts

```diff
@@ -161,4 +210,421 @@ export class Session {
+      return true; // If no identifier is provided, consider it expired
+    }
+
+    const parts = applicationIdentifier.replace(this.KEYLESS_ENVIRONMENT_PREFIX, '').split('_');
+    if (parts.length < 1) {
+      return true; // Invalid format, consider expired
+    }
+
+    const createdDate = parts[0];
+
+    if (!createdDate || createdDate.length < 8) {
+      // Ensure we have at least 4 bytes (8 hex chars)
+      return true; // Invalid timestamp format, consider expired
+    }
+
+    try {
+      const createdDateTimestamp = timestampHexToDate(createdDate);
+      const now = new Date();
+      const diffTimeInHours = differenceInHours(now, createdDateTimestamp);
+
+      if (diffTimeInHours > KEYLESS_RETENTION_TIME_IN_HOURS) {
+        return true;
+      }
+    } catch (error) {
+      this.logger.error({ err: error }, 'Error parsing timestamp');
+
+      // If there's any error parsing the timestamp, consider it expired
+      return true;
+    }
+
+    return false;
+  }
+
+  async processKeyless(): Promise<EnvironmentResponseDto> {
+    if (process.env.NOVU_ENTERPRISE !== 'true') {
+      throw new BadRequestException('Keyless is not supported in community edition');
+    }
+
+    const organization = await this.communityOrganizationRepository.findById(process.env.KEYLESS_ORGANIZATION_ID!);
+
+    if (!organization) {
+      this.logger.error('Keyless Organization not found');
+      throw new InternalServerErrorException('Keyless Organization not found');
+    }
+
+    const user = await this.communityUserRepository.findByEmail(process.env.KEYLESS_USER_EMAIL!);
+
+    if (!user) {
+      throw new InternalServerErrorException('Keyless User not found');
+    }
+
+    const key = `sk_${await this.generateUniqueApiKey.execute()}`;
+    const encryptedApiKey = encryptApiKey(key);
+    const hashedApiKey = createHash('sha256').update(key).digest('hex');
+
+    const encodedDate = dateToTimestampHex(new Date());
+    const identifier = `${this.KEYLESS_ENVIRONMENT_PREFIX}${encodedDate}_${shortId(4)}`;
+    const environment = await this.environmentRepository.create({
+      _organizationId: organization._id,
+      name: `Keyless ${new Date().toISOString()}`,
+      identifier,
+      apiKeys: [
+        {
+          key: encryptedApiKey,
+          _userId: user._id,
+          hash: hashedApiKey,
+        },
+      ],
+    });
+
+    await this.createNovuIntegrationsUsecase.execute(
+      CreateNovuIntegrationsCommand.create({
+        environmentId: environment._id,
+        organizationId: environment._organizationId,
+        userId: user._id,
+        name: 'Keyless Integration',
+        channels: [ChannelTypeEnum.IN_APP],
+      })
+    );
+
+    await this.createWorkflowsUsecase(environment._id, environment._organizationId, user._id);
+
+    return this.convertEnvironmentEntityToDto(environment);
+  }
+
+  async createWorkflowsUsecase(environmentId: string, organizationId: string, userId: string) {
```

Comment:
I think it's ok for now, but I would really want to remove it and add the stateless workflows 😅 Let's chat about it early next week

Reply:
I would love it to be a stateless solution, but at the moment, it does not exist, so I was "forced" to improvise to finish the project.